deepspeed/runtime/zero/stage3.py

```diff
@@ -1159,7 +1159,7 @@ def __reduce_and_partition_ipg_grads(self, safe_mode: bool = False) -> None:
     @instrument_w_nvtx
     def __avg_scatter_contiguous_grads(self, buffer_to_reduce: Tensor) -> List[Tensor]:
         dtype = buffer_to_reduce.dtype
-        if self.communication_data_type == self.dtype:
+        if self.communication_data_type != dtype:
```

Comment:
```suggestion
        if self.communication_data_type != self.dtype:
```

Reply:
I think line 1161 is redundant and could be deleted. What do you think?

Reply:
I agree. Thanks.